pkg/controller/ec2/vpccidrblock/controller.go

```diff
@@ -0,0 +1,210 @@
+	errDisassociate     = "failed to disassociate the VPCCIDRBlock resource"
+)
+
+// SetupVPCCIDRBlock adds a controller that reconciles VPCCIDRBlocks.
+func SetupVPCCIDRBlock(mgr ctrl.Manager, l logging.Logger, rl workqueue.RateLimiter) error {
+	name := managed.ControllerName(v1alpha1.VPCCIDRBlockGroupKind)
+	return ctrl.NewControllerManagedBy(mgr).
+		Named(name).
+		WithOptions(controller.Options{
+			RateLimiter: ratelimiter.NewDefaultManagedRateLimiter(rl),
+		}).
+		For(&v1alpha1.VPCCIDRBlock{}).
+		Complete(managed.NewReconciler(mgr,
+			resource.ManagedKind(v1alpha1.VPCCIDRBlockGroupVersionKind),
+			managed.WithExternalConnecter(&connector{kube: mgr.GetClient(), newClientFn: ec2.NewVPCCIDRBlockClient}),
+			managed.WithReferenceResolver(managed.NewAPISimpleReferenceResolver(mgr.GetClient())),
+			managed.WithConnectionPublishers(),
+			managed.WithInitializers(managed.NewDefaultProviderConfig(mgr.GetClient())),
+			managed.WithLogger(l.WithValues("controller", name)),
+			managed.WithRecorder(event.NewAPIRecorder(mgr.GetEventRecorderFor(name)))))
+}
+
+type connector struct {
+	kube        client.Client
+	newClientFn func(config aws.Config) ec2.VPCCIDRBlockClient
+}
+
+func (c *connector) Connect(ctx context.Context, mg resource.Managed) (managed.ExternalClient, error) {
+	cr, ok := mg.(*v1alpha1.VPCCIDRBlock)
+	if !ok {
+		return nil, errors.New(errUnexpectedObject)
+	}
+	cfg, err := awsclient.GetConfig(ctx, c.kube, mg, cr.Spec.ForProvider.Region)
+	if err != nil {
+		return nil, err
+	}
+	return &external{client: c.newClientFn(*cfg), kube: c.kube}, nil
+}
+
+type external struct {
+	kube   client.Client
+	client ec2.VPCCIDRBlockClient
+}
+
+func (e *external) Observe(ctx context.Context, mgd resource.Managed) (managed.ExternalObservation, error) { // nolint:gocyclo
+	cr, ok := mgd.(*v1alpha1.VPCCIDRBlock)
+	if !ok {
+		return managed.ExternalObservation{}, errors.New(errUnexpectedObject)
+	}
+
+	if meta.GetExternalName(cr) == "" {
+		return managed.ExternalObservation{
+			ResourceExists: false,
+		}, nil
+	}
+
+	response, err := e.client.DescribeVpcsRequest(&awsec2.DescribeVpcsInput{
+		VpcIds: []string{aws.StringValue(cr.Spec.ForProvider.VPCID)},
+	}).Send(ctx)
+	if err != nil {
+		return managed.ExternalObservation{}, awsclient.Wrap(resource.Ignore(ec2.IsVPCNotFoundErr, err), errDescribe)
+	}
+
+	// in a successful response, there should be one and only one object
+	if len(response.Vpcs) != 1 {
+		return managed.ExternalObservation{}, errors.New(errMultipleItems)
+	}
+
+	observed := response.Vpcs[0]
+
+	// update the CRD spec for any new values from provider
+	current := cr.Spec.ForProvider.DeepCopy()
+	ec2.LateInitializeVPCCIDRBlock(&cr.Spec.ForProvider, &observed)
+
+	currentStatusCode, err := ec2.FindVPCCIDRBlockStatus(meta.GetExternalName(cr), observed)
+	if err != nil {
+		return managed.ExternalObservation{}, errors.Wrap(resource.Ignore(ec2.IsCIDRNotFound, err), errDescribe)
```

Comment:
```suggestion
		return managed.ExternalObservation{}, errors.Wrap(err, errDescribe)
```

This is actually a valid error case for this controller. We can ignore `NotFound` error only if we are the one creating the resource, in this case `IsCIDRNotFound` does that.